### ISSUE TITLE
TFDS object detection to coco format

### ISSUE THREAD

**Issue by davidslater** [twosixlabs/armory#1813]
The object detection datasets are not in standard COCO format. We should move them to output in standard COCO format.
This file is a good model:
https://github.com/tensorflow/datasets/blob/master/tensorflow_datasets/object_detection/coco.py

See here for the spec (under 1. Object Detection):
https://cocodataset.org/#format-data

We will want a format similar to:
```
'objects':
              tfds.features.Sequence({
                  'id': np.int64,  # coco has unique ID for each detection. We may omit this field, I think.
                  'image_id': np.int64,
                  'area': np.int64,
                  'bbox': tfds.features.BBoxFeature(),
                  'label': tfds.features.ClassLabel(num_classes=80),  # this maps down to the set of categories actually used
                  'iscrowd': np.bool_,
              }),
```
This requires ensuring that each dict only is a single detection.

Currently, some datasets have different names (e.g., xview uses `boxes` instead of `bbox`, carla datasets also, I think).
Also, some datasets (xview again) have a dict of arrays instead of a sequence of dicts. 

We should be very explicit about what we expect / allow for use in Armory, and be consistent.

**Comment by lcadalzo** [comment]
From discussion with @davidslater: the intention is to continue having raw datasets build in the [tf format ](https://www.tensorflow.org/datasets/api_docs/python/tfds/features/BBoxFeature) of normalized `[y1, x1, y2, x2]`, but instead of converting to pytorch format of unnormalized `[x1, y1, x2, y2]` (which is what we do currently), we'll convert to the [coco forma](https://cocodataset.org/#format-data)t of unnormalized `[x1, y1, width, height]`
